dattri/algorithm/data_shapley.py

```diff
@@ -0,0 +1,178 @@
+"""This module implement data shapley valuation."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+    from typing import List, Optional, Tuple
+
+    from dattri.task import AttributionTask
+
+import torch
+from tqdm import tqdm
+
+from .base import BaseAttributor
+from .utils import _check_shuffle
+
+
+def default_dist_func(
+        batch_x: Tuple[torch.Tensor],
+        batch_y: Tuple[torch.Tensor],
+) -> torch.Tensor:
+    """Default distance function for KNN.
+
+    Args:
+        batch_x (Tuple[torch.Tensor]): The batch to be calculated
+            distances on. The embeddings is default to be the
+            first element.
+        batch_y (Tuple[torch.Tensor]): The batch to be calculated
+            distances on. The embeddings is default to be the
+            first element.
+
+    Returns:
+        Tensor: The calculated Euclidean distance.
+    """
+    coord1 = batch_x[0]
```

Comment:
We are assuming the dataloader will yield batches in the form of (x, y) pair here.

In this case, maybe we could remove train_labels and test_labels in the self.attribute() function, and add a couple of value errors if the dataloader does not follow this format.

Reply:
If we follow this, then one of my above concern can also be resolved.

Reply:
I have updated by making the labels optional.